packages/npm-dependencies-validation/src/types.ts

```diff
@@ -1,30 +1,27 @@
-export type NPMIssueType = "missing" | "invalid" | "extraneous";
+import { OutputChannel } from "vscode";
 
-export type Version = "required" | "version";
-
-export type NPMDependencyIssue = {
-  name: string;
+export type NpmLsResult = {
+  name?: string;
   version?: string;
-  type: NPMIssueType;
-  devDependency?: boolean;
+  problems: string[];
 };
 
-export type NpmLsRDependencies = {
+export type DependenciesPropertyName = "dependencies" | "devDependencies";
+
+export type PackageJson = {
   name: string;
   version: string;
-  dependencies: Dependencies;
-};
+} & DependenciesProperties;
 
-export type Dependencies = {
-  [key: string]: Dependency;
+export type DependenciesProperties = {
```

Comment:
This seems a bit complex, could we just declare everything directly on PackageJson type?
I could not find where else are using these sub-types that are used to assemble `PackageJson `

Reply:
Using this syntax I removed code duplication.
Currently we do not need all these properties, but we will probably need them in the future.
Should I remove them for now?